nautobot_ssot/integrations/infoblox/diffsync/adapters/infoblox.py

```diff
@@ -307,7 +307,31 @@ def _load_dns_host_record_for_ip(self, ref: str, ip_record: object, namespace: s
             ip_record (object): Parent IP Address record
             namespace (str): Namespace of this record
         """
-        host_record = self.conn.get_host_record_by_ref(ref)
+        try:
+            host_record = self.conn.get_host_record_by_ref(ref)
+        except requests.exceptions.HTTPError as exc:
+            if exc.response.status_code == 404:
```

Comment:
I'll say same thing to you as I did @MeganerdDev, this needs to be behind a setting for the integration where the user explicitly opts into this behavior. Following principle of least astonishment we don't want the users surprised by records disappearing that they don't expect.

Reply:
Also, do you think you could write a unit test to cover this?

Reply:
I'm not aware of a method of excluding DHCP pools but I could just be missing it.